### PR TITLE
OTA, Update Releasing Notes, & Enforce Version Pattern

### LINKED ISSUE #38: auto_firmware_version.py not safe/reliable
The` auto_firmware_version.py` script is not safe way to establish the version.  The script presumes that` git describe --tags` will work, but that requires that the git repository actually have tags.  Without it you get:
```
fatal: No names found, cannot describe anything.
```
And the firmware version embedded into the source is empty.  This matters because anyone who forks the repository will not carry over all the tags, each repository have their own tag history.

A better place to pull the version number from would be `manifest.json` file.  This contains the actual version number and is independent of any git tagging.

### PR DESCRIPTION
**Pull request by donavanbecker**
## Update Releasing Notes & Enforce Version Pattern
***(1st Commit)***
Added more details to the releasing.md and also checked for the pattern of the release tag. If it doesn't follow the pattern then the workflow will stop

## Update Description for Each Web Server Action Button:
***(2nd Commit)***
**Wait till these PRs are merged:**

- [x] [Add HTTPUpdateServer OTA support #50](https://github.com/ratgdo/homekit-ratgdo/pull/50) conflicts resolved with [#72](https://github.com/ratgdo/homekit-ratgdo/pull/72)
- [x] [Remove duplicate SSIDs from setup web page #53](https://github.com/ratgdo/homekit-ratgdo/pull/53)
- [x] [Get version from manifest file and QR code from svg. fixes #38 and #45 #44](https://github.com/ratgdo/homekit-ratgdo/pull/44)

Screenshot 2024-01-01 at 10 22 55 PM

### CODE REVIEW

File: docs/releasing.md
Comment (BEFH) [inline docs/releasing.md:1]: Missing a `+` in the regex on line 8

Comment (donavanbecker) [reply docs/releasing.md:1]: This will be changing, still trying to work this out.

**Comment by iyerusad** [comment]
Was able to deploy a build from this forked release: https://github.com/donavanbecker/homekit-ratgdo/releases/tag/v0.14.0

Seems basic functionality is working, door open/close, light, obstruction sensing. Didn't test OTA, but presumably can if a rolled up release gets pushed on the fork.

**Comment by iyerusad** [comment]
Successfully updated OTA via web interface from forked release: https://github.com/donavanbecker/homekit-ratgdo/releases/tag/v0.14.0 to new latest release: https://github.com/ratgdo/homekit-ratgdo/releases/tag/v0.8.0

Looks like basic OTA functionality is now merged in, woot woot.

**Pull request closed without merging**